models/mnist_gan_graph.py

```diff
@@ -0,0 +1,232 @@
+        )
+
+        # saver = tf.train.Saver(max_to_keep=1)
+
+    def load_data(self):
+        # Read the input data
+        # In this case, MNIST + batch and shuffle it. In our case, it will be quite different.
+
+        from tensorflow.examples.tutorials.mnist import input_data
+
+        def read_data_sets(data_dir):
+            """Returns training and test tf.data.Dataset objects."""
+            data = input_data.read_data_sets(data_dir, one_hot=True)
+            # train_ds = tf.data.Dataset.from_tensor_slices((data.train.images,
+            #                                               data.train.labels))
+            # test_ds = tf.data.Dataset.from_tensors(
+            #   (data.test.images, data.test.labels))
+            return (data.train, data.test)
+
+        device, data_format = ('/gpu:0', 'channels_first')
+        if FLAGS.training.no_gpu:
+            device, data_format = ('/cpu:0', 'channels_last')
+        print('Using device %s, and data format %s.' % (device, data_format))
+
+        # Load the datasets
+        train_ds, test_ds = read_data_sets(FLAGS.data.in_dir)
+        # train_ds = train_ds.shuffle(60000).batch(FLAGS.batch_size)
+        return shuffle(train_ds.images, train_ds.labels)
+
+    def train_batch(self, batch):
+        BATCH_SIZE = FLAGS.model.optimization.batch_size
+
+        # 1. Train Discriminator
+        # X_batch = images_to_vectors(batch.permute(0, 2, 3, 1).numpy())
+        batch_noise = noise((BATCH_SIZE, self.NOISE_SIZE))
+        feed_dict = {self.images_input: batch, self.noise_input: batch_noise}
+        d_error, _ = self.session.run([self.d_loss, self.d_opt], feed_dict=feed_dict)
+
+        # 2. Train Generator
+        feed_dict = {self.noise_input: batch_noise}
+        g_error, _ = self.session.run([self.g_loss, self.g_opt], feed_dict=feed_dict)
+
+        return d_error, g_error
+
+    # Generate images from test noise
+    def test_eval(self, epoch_info, test_noise_random, test_noise_interpolated):
+        self.session.run(
+            self.IMAGE_SUMMARIES,
+            feed_dict={
+                self.noise_input: test_noise_random,
+                self.noise_input_interpolated: test_noise_interpolated
+            }
+        )
+
+    def run(self):
+        BATCH_SIZE = FLAGS.model.optimization.batch_size
+        train_X, train_Y = self.load_data()
+
+        num_test_samples = 36
+        test_noise_random = noise((num_test_samples, self.NOISE_SIZE))
+        test_noise_interpolated = noise((num_test_samples, self.NOISE_SIZE), dist='linspace')
+
+        # Iterate through epochs
+        for epoch in range(FLAGS.model.optimization.epochs):
+            print("Epoch %d" % epoch, flush=True)
+            for n_batch, batch in enumerate(next_batch(train_X, BATCH_SIZE)):
+                d_error, g_error = self.train_batch(batch)
+
+                # Test noise
+                if n_batch % 500 == 0:
```

Comment:
I guess this should be a flag parameter or something.

Reply:
Fix postponed to #6